airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2849,7 +2849,7 @@ def next(self) -> list | None:
                 return None
 
             query_results = self._get_query_result()
-            if "rows" in query_results and query_results["rows"]:
+            if query_results.get("rows"):
```

Comment:
```suggestion
            if rows := query_results.get("rows"):
```
And remove assign from L2856